strawberryfields/api/connection.py

```diff
@@ -0,0 +1,284 @@
+# Copyright 2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+TODO
+"""
+from datetime import datetime
+import io
+import json
+import logging
+from typing import List
+
+import numpy as np
+import requests
+
+from strawberryfields.configuration import DEFAULT_CONFIG
+from strawberryfields.io import to_blackbird
+from strawberryfields.program import Program
+from .job import Job, JobStatus
+from .result import Result
+
+log = logging.getLogger(__name__)
+
+
+class RequestFailedError(Exception):
+    """Raised when a request to the remote platform returns an error response."""
+
+
+class Connection:
+    """Manages remote connections to the remote job execution platform and exposes
+    various job operations.
+
+    For basic usage, it is not necessary to manually instantiate this object; the user
+    is encouraged to use the higher-level interface provided by
+    :class:`~strawberryfields.engine.StarshipEngine`.
+
+    **Example:**
+
+    The following example instantiates a :class:`~strawberryfields.api.Connection` for a
+    given API authentication token, tests the connection, submits a new job, and
+    retrieves an existing job.
+
+    >>> connection = Connection(token="abc")
+    >>> success = connection.ping() # `True` if successful, `False` if the connection fails
+    >>> job = connection.create_job("chip2", program, shots=123)
+    >>> job
+    <Job: id=d177cbf5-1816-4779-802f-cef2c117dc1a, ...>
+    >>> job.status
+    <JobStatus.QUEUED: 'queued'>
```

Comment:
Users won't likely ever use a `Connection` directly, but they will certainly interact with a `Job`. The string printed here could be improved to be simpler and less redundant, e.g., `<JobStatus: queued>`

Reply:
Another thought - if the user wants to programmatically check job status, they will need to import the `JobStatus` enum from within the API. Will they also be able to compare via a simple string? i.e., `job.status == "queued"`?

Reply:
They can compare to a string with `job.status.value == "queued"`. It's not ideal, but since the alternative is dropping the enum entirely, I feel that it's a decent tradeoff :thinking: 

I guess the options here are:
- Leave it as is
- Override the `__eq__` method to enable comparison with a string, i.e. `JobStatus.QUEUED == "queued"` -> `True`
- Add properties on `JobStatus`, e.g. `is_queued`, `is_completed`, etc.
- Get rid of the enum and replace all usages with string literals

What do you think?

Reply:
Actually, please ignore the above.

After more thought, the obvious solution (to me) would be to keep `JobStatus` internally but expose it as a string via the property. I will make that change - let me know if you think there is a better way!